src/nvim/api/win_config.c

```diff
@@ -240,6 +247,49 @@ void nvim_win_set_config(Window window, Dict(float_config) *config, Error *err)
   }
 }
 
+Dictionary config_put_bordertext(Dictionary config, FloatConfig *fconfig, BorderTextType bordertext_type)
+{
+  VirtText chunks;
+  AlignTextPos align;
+  char* field_name;
+  char* field_pos_name;
+  if (bordertext_type == kBorderTextTitle) {
+    chunks = fconfig->title_chunks;
+    align = fconfig->title_pos;
+    field_name = "title";
+    field_pos_name = "title_pos";
+  } else if (bordertext_type == kBorderTextFooter) {
+    chunks = fconfig->footer_chunks;
+    align = fconfig->footer_pos;
+    field_name = "footer";
+    field_pos_name = "footer_pos";
+  }
+
+  Array bordertext = ARRAY_DICT_INIT;
+  for (size_t i = 0; i < chunks.size; i++) {
+    Array tuple = ARRAY_DICT_INIT;
+    ADD(tuple, CSTR_TO_OBJ(chunks.items[i].text));
+    if (chunks.items[i].hl_id > 0) {
+      ADD(tuple, CSTR_TO_OBJ(syn_id2name(chunks.items[i].hl_id)));
+    }
+    ADD(bordertext, ARRAY_OBJ(tuple));
+  }
+  PUT(config, field_name, ARRAY_OBJ(bordertext));
+
+  char *pos;
+  if (align == kAlignLeft) {
+    pos = "left";
+  } else if (align == kAlignCenter) {
+    pos = "center";
+  } else {
+    pos = "right";
+  }
```

Comment:
Personally I prefer to use `switch` statements for enums. This is (in my opinion) more semantically clear, and also has the benefit that if a new enum variant is ever added, the compiler will warn you if the switch is non-exhaustive.

```suggestion
  switch (align) {
  case kAlignLeft:
    pos = "left";
    break;
  case kAlignCenter:
    pos = "center";
    break;
  case kAlignRight:
    pos = "right";
    break;
  }
```

(I know you didn't write this code, you just moved it, but that's as good a time as any to make minor changes like this).